### PR TITLE
Exposing AttributeSet::dropAttributes via the PointDataLeafNode

### PR DESCRIPTION
**Pull request by jbrd**
This is related to my previous pull request (exposing copyAttributeValues), but more general and useful I think. In the future we'll be adding the point sort and merge algorithms, as well as the Points Attribute SOP - these rely on being able to create and remove attributes from an existing PointDataLeafNode...

### CODE REVIEW

**Comment by jbrd** [comment]
Thanks for the feedback Dan. I can attend to both of those...

**Comment by jbrd** [comment]
I've pushed up some changes that expose the expected/replace variant of dropAttributes instead, and added a unit test. I'd like to iterate on the wrapper with you internally a bit before we push this to github, for now, this is all the changes we need in the public repo to move forward...

File: openvdb_points/tools/PointDataGrid.h [hunk 1]
```diff
@@ -328,6 +328,15 @@ class PointDataLeafNode : public tree::LeafNode<T, Log2Dim> {
         mAttributeSet->appendAttribute(attribute, expected, replacement);
     }
 
+    /// @brief Drop attributes with @a pos indices.
+    /// Requires the current descriptor to match @a expacted
```
Comment (danrbailey) [inline openvdb_points/tools/PointDataGrid.h]: typo - "expected"

File: openvdb_points/unittest/TestPointDataLeaf.cc [hunk 2]
```diff
@@ -845,6 +847,55 @@ TestPointDataLeaf::testIO()
 }
 
 
+void TestPointDataLeaf::testDropAttributes()
```
Comment (danrbailey) [inline openvdb_points/unittest/TestPointDataLeaf.cc]: This is quite a lot of code to test a function which is really just a wrapper. Although it means not achieving 100% code coverage, I suggest we rely on the TestAttributeSet::testAttributeSet() function to test appending and dropping attributes from the attribute set to minimise the extra code duplication.

**Pull request merged**